tools/buildmgr/cbuild/src/CbuildModel.cpp

```diff
@@ -735,64 +735,41 @@ const bool CbuildModel::EvalDeviceName() {
   return true;
 }
 
-set<string> CbuildModel::SplitArgs(const string& args, const string& delim, bool relativePath) {
+list<string> CbuildModel::SplitArgsList(const string& args, const string& delim, bool relativePath) {
   /*
   SplitArgs:
   Split arguments from the string 'args' into a set of strings
   If 'relativePath' is true relative paths are normalized
   */
 
-  set<string> s;
+  list<string> s;
   size_t end = 0, start = 0, len = args.length();
   while (end < len) {
     if ((end = args.find(delim, start)) == string::npos) end = len;
     string flag = args.substr(start, end - start);
     if (relativePath)
       flag = CbuildUtils::StrPathAbsolute(flag, m_prjFolder);
-    s.insert(flag);
-    start = end + 1;
-  }
-  return s;
-}
-
-list<string> CbuildModel::SplitArgsList(const string& args) {
-  /*
-  SplitArgsList:
-  Split arguments from the string 'args' into a list of strings
-  Normalize relative paths
-  */
-
-  list<string> s;
-  size_t end = 0, start = 0, len = args.length();
-  while (end < len) {
-    if ((end = args.find(" -", start)) == string::npos) {
-      end = len;
-    }
-    string flag = args.substr(start, end - start);
-    flag = CbuildUtils::StrPathAbsolute(flag, m_prjFolder);
     s.push_back(flag);
     start = end + 1;
   }
   return s;
 }
 
-set<string> CbuildModel::MergeArgs(const set<string>& add, const set<string>& remove, const set<string>& reference) {
+list<string> CbuildModel::MergeArgsList(const list<string>& add, const list<string>& remove, const list<string>& reference) {
   /*
   MergeArgs:
    - Merge 'add' arguments from 'reference' avoiding duplicates
    - Remove 'remove' items from 'reference'
   */
 
-  set<string> list = reference;
+  list<string> list = reference;
   for (auto add_item : add) {
-    list.insert(add_item);
+    list.push_back(add_item);
   }
   for (auto rem_item : remove) {
-    for (auto ref_item : list) {
-      if (rem_item.compare(ref_item) == 0) {
-        list.erase(ref_item);
-        break;
-      }
+    auto first_match = std::find_if(list.cbegin(), list.cend(), [rem_item](auto i){ return i == rem_item; });
```

Comment:
It requires `#include <algorithm>`

Reply:
Ok, so if I understand correctly the PR needs to also fix
* RTE Component order processing should be changed to reflect the order specified from the CPRJ?
* PR should also be amended to use list<> instead of set<> https://github.com/Open-CMSIS-Pack/devtools/blob/main/tools/buildmgr/cbuildgen/include/BuildSystemGenerator.h#L91 
* #include <algorithm>
* Update tests

Reply:
Right but please leave changes in the `rtemodel` for a separate PR for better review and evaluation.